### PR TITLE
Migrate bnb trades

### PR DESCRIPTION
**Pull request by YordanPavlov**
A migration of [BNB trades exporter](https://github.com/santiment/bnb-exporter-trades) into the common exporter repository. A lot of code was reused from the already migrated [BNB transactions exporter](https://github.com/santiment/bnb-exporter).

### CODE REVIEW

**Review by apozdeyev — APPROVED** [review]

File: bin/run_bnb_trades.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ #! /bin/sh
+
+BLOCKCHAIN=bnb NODE_URL=https://dex.binance.org/api/v1/trades/ BNB_MODE=trades \
+docker-compose -f docker/docker-compose.yaml up --build && docker-compose -f docker/docker-compose.yaml rm -f
```
Comment (WonderBeat) [inline bin/run_bnb_trades.sh:4]: what do you think of `set -e`. So you can avoid `&&` here

File: blockchains/bnb/lib/bnb_transactions_fetcher.js [hunk 2]
```diff
@@ -59,7 +60,11 @@ class BNBTransactionsFetcher {
       return nextRange
     }
 
-    this.lastBlockTimestamp = await getLastBlockTimestamp(metrics);
+    this.lastBlockTimestamp = this.bnbTradesMode ?
+    await getLastTradesBlockTimestamp(metrics)
+    :
```
Comment (WonderBeat) [inline blockchains/bnb/lib/bnb_transactions_fetcher.js:65]: Dangling `:` looks strange. Can you pad it a little bit?
And btw, do you use any prettyfier in JS?

File: blockchains/bnb/lib/fetch_transactions.js [hunk 3]
```diff
@@ -29,21 +43,35 @@ function sendTrxQuery(trxId, queue, metrics) {
...
-async function fetchTimeInterval(queue, startTimeMsec, endTimeMsec, nodeResponsePromises, metrics) {
+function getNumPagesToIterate(firstPageResult, startTimeMsec, endTimeMsec, bnbTradesMode) {
+  if (typeof firstPageResult === 'undefined' ||
+   (bnbTradesMode && firstPageResult.trade === 'undefined') ||
+   (!bnbTradesMode && firstPageResult.txArray === 'undefined')) {
+      throw ("Error in fetch interval ", startTimeMsec, " - ", endTimeMsec)
+  }
+
+  return bnbTradesMode ?
+  Math.ceil(firstPageResult.total / constants.NUM_TRADE_ROWS_FETCH)
```
Comment (WonderBeat) [inline blockchains/bnb/lib/fetch_transactions.js:54]: This additional parameter that we pass everywhere, oh, I wish we can avoid it. 
Can this switch be expressed with OOP somehow?
And this function in general? Why does it has so many arguments if it requires only one. The rest of params are only used to check preconditions, right? Can we check preconditions in another place? Current function signature confuses me a bit. It should "fetch time interval", but actually it just calculates the number of 'pages' in `firstPageResult`

File: blockchains/bnb/lib/fetch_transactions.js [hunk 4]
```diff
@@ -188,9 +216,29 @@ function filterRepeatedTransactions(listTrx) {
   }
 }
 
+/**
+ * For reasons currently unknown the BNB API returns same results on different pages. Last result from page x may
+ * appear as first result on page x + 1. That is the oldest result on page x may be the newest on page x + 1.
+ * @param {*} listTrx An array of trades to be filtered
+
+ */
```
Comment (WonderBeat) [inline blockchains/bnb/lib/fetch_transactions.js:224]: Do you have a testcase for this?
Does it only work for sequential duplicates?

**Review by WonderBeat — COMMENTED** [review]
LGTM, though this additional parameters increases overal code complexity

File: blockchains/bnb/lib/fetch_transactions.js (hunk 3 above)
Comment (YordanPavlov) [reply blockchains/bnb/lib/fetch_transactions.js]: The function is stateless even though the arguments are indeed a lot. I am not sure OOP is simpler to grasp compared to procedural programming. A function that has no state and acts only upon arguments, even if they are more than 2 as per OOP rules is fine, no? I have broken it into a `check` part and main computation to simplify the logic.

File: bin/run_bnb_trades.sh (hunk 1 above)
Comment (YordanPavlov) [reply bin/run_bnb_trades.sh]: Sounds good. Done.

**Comment by WonderBeat** [comment]
:+1:

**Pull request merged**